src/lsm/lsm_cursor.c

```diff
@@ -1458,6 +1458,12 @@ __clsm_put(WT_SESSION_IMPL *session, WT_CURSOR_LSM *clsm,
 	if (position)
 		clsm->current = primary;
 
+	/* Reserve: only update the primary chunk, ignore the record count. */
+	if (reserve) {
+		primary->set_key(primary, key);
+		return (primary->reserve(primary));
+	}
```

Comment:
More specifically, I think reserve should exactly replace the `update` in the loop below and we should continue to call `insert` as we do for updates, because this `insert` is special in that it is only doing a conflict check in overlapping chunks (usually `nupdates == 1`).